app/main/controllers/activity/RTMediaBuddyPressActivity.php

```diff
@@ -64,6 +67,81 @@ function __construct() {
 		}
 	}
 
+	/**
+	 * Show media even if the text is long with read more option.
+	 *
+	 * @param string $excerpt  Excerpt of the activity text.
+	 * @param string $text     Actual text of activity.
+	 * @param string $readmore Read more text.
+	 *
+	 * @return string Custom excerpt if conditions are match.
+	 */
+	public function bp_activity_truncate_entry( $excerpt, $text, $readmore ) {
+		global $activities_template;
+
+		$excerpt_length = bp_activity_get_excerpt_length();
+		// Run the text through the excerpt function. If it's too short, the original text will be returned.
+		$temp_excerpt = bp_create_excerpt( $text, $excerpt_length, array() );
+		if ( strlen( $temp_excerpt ) >= strlen( strip_shortcodes( $text ) ) ) {
+			return $excerpt;
+		}
+
+		// Get current activity id.
+		$activity_id = bp_get_activity_id();
+
+		// We need to separate text and rtMedia images, for this we need DOM manipulation.
+		$dom = new DOMDocument();
+		$dom->loadHTML( $text );
+		// We need to find div having rtmedia-activity-text class, but no direct method for it.
+		// So we need to iterate.
+		$div_list = $dom->getElementsByTagName( 'div' );
+
+		// We're storing first div to create final markup.
+		// If we create markup from dom object, it'll create whole HTML which we don't want.
+		$first_div = '';
+
+		foreach ( $div_list as $div ) {
+			// Set first div.
+			if ( empty( $first_div ) ) {
+				$first_div = $div;
+			}
+
+			// We need div with class attribute.
+			if ( empty( $div->attributes ) ) {
+				continue;
+			}
+
+			$atts = $div->attributes;
+			// Check attributes by iterating them.
+			foreach ( $atts as $att ) {
```

Comment:
Check !empty($atts)

Reply:
This is already checked on line 114 `if ( empty( $div->attributes ) ) {`